src/plugins/login/utils/simpleFileTokenCache.ts

```diff
@@ -0,0 +1,92 @@
+import fs from "fs";
+import path from "path";
+import os from "os";
+import * as adal from "adal-node";
+
+const CONFIG_DIRECTORY = path.join(os.homedir(), ".azure");
+const DEFAULT_SLS_TOKEN_FILE = path.join(CONFIG_DIRECTORY, "slsTokenCache.json");
+
+export class SimpleFileTokenCache implements adal.TokenCache {
+  private entries: any[] = [];
+  private subscriptions: any[] = [];
+
+  public constructor(private tokenPath: string = DEFAULT_SLS_TOKEN_FILE) {
+    this.load();
+  }
+
+  public add(entries: any, cb?: any) {
```

Comment:
what's `cb`? a more descriptive name would be better

Reply:
callback

Reply:
it's a callback, I'll switch the name